apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableClientPartial.cs

```diff
@@ -46,19 +48,45 @@ public abstract partial class BigtableClient
         /// "default" application profile will be used.
         /// </param>
         /// <returns>The task representing the created <see cref="BigtableClient"/>.</returns>
-        public static async Task<BigtableClient> CreateAsync(ServiceEndpoint endpoint = null, BigtableServiceApiSettings settings = null, string appProfileId = null)
+        public static async Task<BigtableClient> CreateAsync(BigtableServiceApiSettings settings = null, ClientCreationSettings clientCreationSettings = null, string appProfileId = null)
         {
-            var client = await BigtableServiceApiClient.CreateAsync(endpoint, settings).ConfigureAwait(false);
-            return new BigtableClientImpl(client, appProfileId) { UnderlyingClientSettings = settings };
+            clientCreationSettings?.Validate();
```

Comment:
It might be worth trying to generalize all this with Pubsub. Not in this PR, but can you add a TODO above this so we can remember to look into it? Thanks.